aarch64-dit/src/lib.rs

```diff
@@ -6,14 +6,92 @@
 )]
 #![warn(missing_docs, rust_2018_idioms, unused_qualifications)]
 
+//! ## Usage
+//!
+//! ```
+//! use aarch64_dit::Dit;
+//!
+//! let dit = Dit::init();
+//! assert!(!dit.is_enabled());
+//! let _guard = dit.enable();
+//! assert!(dit.is_enabled());
+//! ```
+
 #[cfg(not(target_arch = "aarch64"))]
 compile_error!("This crate only builds on `aarch64` targets");
 
 use core::arch::asm;
 
+cpufeatures::new!(dit_supported, "dit");
+
+/// Data-Independent Timing: support for enabling features of AArch64 CPUs which improve
+/// constant-time operation.
+pub struct Dit {
+    supported: dit_supported::InitToken,
+}
+
+impl Dit {
+    /// Initialize Data-Independent Timing using runtime CPU feature detection.
+    pub fn init() -> Self {
+        Self {
+            supported: dit_supported::init(),
+        }
+    }
+
+    /// Enable Data-Independent Timing (if available).
+    ///
+    /// Returns an RAII guard that will return DIT to its previous state when dropped.
+    #[must_use]
+    pub fn enable(&self) -> Guard<'_> {
+        let was_enabled = if self.is_supported() {
+            let was_enabled = unsafe { get_dit_enabled() };
+            unsafe { set_dit_enabled() };
```

Comment:
This will result in two separate function calls because `#[target_feature(enable = "dit")]` prevents inlining. It may be worth to tweak `set_dit_enabled` to return DIT state flag and write here `let was_enabled = unsafe { set_dit_enabled() };`